WordPressCom-Stats-iOS/WPStatsService.m

```diff
@@ -173,6 +204,90 @@ - (void)retrieveAllStatsForDate:(NSDate *)date
 }
 
 
+- (void)retrieveInsightsStatsWithAllTimeStatsCompletionHandler:(StatsAllTimeCompletion)allTimeCompletion
+                                     insightsCompletionHandler:(StatsInsightsCompletion)insightsCompletion
+                                 todaySummaryCompletionHandler:(StatsSummaryCompletion)todaySummaryCompletion
+                                                 progressBlock:(void (^)(NSUInteger numberOfFinishedOperations, NSUInteger totalNumberOfOperations)) progressBlock
+                                   andOverallCompletionHandler:(void (^)())overallCompletionHandler
+{
+    NSString *cacheKey = @"BatchInsights";
+    NSString *allTimeCacheKey = @"AllTime";
+    NSString *insightsCacheKey = @"Insights";
+    NSString *todayCacheKey = @"Today";
+    
+    NSMutableDictionary *cacheDictionary = [self.ephemory objectForKey:cacheKey];
+    if (cacheDictionary.count == 3) {
```

Comment:
This one is a little bit hard to follow. Can we replace the `3` with a constant?